test/helper.py

```diff
@@ -252,7 +266,13 @@ def expect_info_dict(self, got_dict, expected_dict):
 
     test_info_dict = sanitize_got_info_dict(got_dict)
 
-    missing_keys = set(test_info_dict.keys()) - set(expected_dict.keys())
+    # Check for invalid/misspelled field names being returned by the extractor
+    invalid_keys = sorted(set(test_info_dict.keys()) - set(ALLOWED_KEYS_SORT_ORDER))
+    self.assertFalse(invalid_keys, f'Invalid fields returned by the extractor: {", ".join(invalid_keys)}')
+
+    missing_keys = sorted(
+        set(test_info_dict.keys()) - set(expected_dict.keys()),
```

Comment:
```suggestion
        test_info_dict.keys() - expected_dict,
```
or
```suggestion
        test_info_dict.keys() - expected_dict.keys(),
```

Reply:
i like explicitly using `.keys()` for both